src/plugins/crypto.c

```diff
@@ -1507,29 +1575,67 @@ gboolean bd_crypto_luks_remove_key_blob (const gchar *device, const guint8 *pass
  * Tech category: %BD_CRYPTO_TECH_LUKS-%BD_CRYPTO_TECH_MODE_REMOVE_KEY
  */
 gboolean bd_crypto_luks_remove_key (const gchar *device, const gchar *pass, const gchar *key_file, GError **error) {
+    struct crypt_device *cd = NULL;
+    gint ret = 0;
+    guint64 progress_id = 0;
+    gchar *msg = NULL;
     gboolean success = FALSE;
     gchar *key_buf = NULL;
     gsize buf_len = 0;
 
+    msg = g_strdup_printf ("Started removing key from the LUKS device '%s'", device);
+    progress_id = bd_utils_report_started (msg);
+    g_free (msg);
+
+    ret = crypt_init (&cd, device);
+    if (ret != 0) {
+        g_set_error (error, BD_CRYPTO_ERROR, BD_CRYPTO_ERROR_DEVICE,
+                     "Failed to initialize device: %s", strerror_l (-ret, c_locale));
+        bd_utils_report_finished (progress_id, (*error)->message);
```

Comment:
Probably better to use the `l_error` local intermediary result variable so that `error` can be passed in as `NULL`.

Alternatively, since this patter is repeating quite often, might be useful to think about some local `bd_` wrapper for the related job reporting. Just an idea, could be a macro even.

Reply:
Oops, forgot to fix it here.